pkgs/build-support/dotnet/build-dotnet-module/hooks/dotnet-validate-lockfile.sh

```diff
@@ -0,0 +1,70 @@
+# Validate a lockfile does not depend on any floating Nuget dependencies
+# Note that all ranges other than floating resolve to the lowest available version by default
+# See https://docs.microsoft.com/en-us/nuget/concepts/package-versioning#version-ranges for more information
+dotnetValidateLockfileHook() {
+    echo "Executing dotnetValidateLockfileHook"
+
+    if [[ ! -f "${lockfilePath-}" ]]; then
+        echo "Could not locate the lockfile! Consider setting \"dontDotnetValidateLockfile\""
+        exit 1
+    fi
+
+    lockfileVersion="$(jq ".version" < "${lockfilePath}")"
+
+    if (( "${lockfileVersion}" != 1 )); then
```

Comment:
Newer dotnet (at least 6.0.7) generate lockfiles with version 2. I didn't encounter any changes that break this script, so you may consider extending the version check here.

```suggestion
    if (( "${lockfileVersion}" != 1 && "${lockfileVersion}" != 2 )); then
```

Reply:
Done, thanks for the review!